server/middleware/authentication.js

```diff
@@ -364,3 +374,28 @@ export function mustBeLoggedIn(req, res, next) {
     }
   });
 }
+
+export const checkTwoFactorAuthJWT = (req, res, next) => {
+  parseJwtNoExpiryCheck(req, res, e => {
```

Comment:
Let's simplify and use directly `jwt.verify` instead of this `parseJwtNoExpiryCheck` + `checkJwtExpiry` which are a bit weird.